src/test/java/com/savvato/tribeapp/services/ReviewSubmittingUserServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith({SpringExtension.class})
+public class ReviewSubmittingUserServiceImplTest extends
+        AbstractServiceImplTest {
+
+    @TestConfiguration
+    static class ReviewSubmittingUserServiceTestContextConfiguration {
+
+        @Bean
+        public ReviewSubmittingUserService reviewSubmittingUserService() {
+            return new ReviewSubmittingUserServiceImpl();
+        }
+    }
+
+    @Autowired
+    ReviewSubmittingUserService reviewSubmittingUserService;
+
+    @MockBean
+    ToBeReviewedRepository toBeReviewedRepository;
+
+    @MockBean
+    ReviewSubmittingUserRepository reviewSubmittingUserRepository;
+
+    // Test that the method returns the expected DTO versions of the ToBeReviewed phrases associated with a user
+    @Test
+    public void testGetUserPhrasesToBeReviewedHappyPath() {
+        List<Long> toBeReviewedIds = new ArrayList<>(Arrays.asList(1L, 2L));
+        Mockito.when(reviewSubmittingUserRepository.findToBeReviewedIdByUserId(anyLong
+                ())).thenReturn(toBeReviewedIds);
+
+        String testWord = "test";
+        List<ToBeReviewed> tbrs = new ArrayList<>();
+        for (Long toBeReviewedId : toBeReviewedIds) {
+            ToBeReviewed tbr = new ToBeReviewed();
+            tbr.setId(toBeReviewedId);
+            tbr.setHasBeenGroomed(true);
+            tbr.setAdverb(testWord);
+            tbr.setVerb(testWord);
+            tbr.setPreposition(testWord);
+            tbr.setNoun(testWord);
+            tbrs.add(tbr);
+        }
+
+        Mockito.when(toBeReviewedRepository.findById(anyLong()))
+                .thenReturn(Optional.of(tbrs.get(0)))
+                .thenReturn(Optional.of(tbrs.get(1)));
+
+        List<ToBeReviewedDTO> expectedDTOs = new ArrayList<>();
+        for (ToBeReviewed tbr : tbrs) {
+            ToBeReviewedDTO tbrDTO = ToBeReviewedDTO.builder().build();
+            tbrDTO.id = tbr.getId();
+            tbrDTO.hasBeenGroomed = tbr.isHasBeenGroomed();
+            tbrDTO.adverb = tbr.getAdverb();
+            tbrDTO.verb = tbr.getVerb();
+            tbrDTO.preposition = tbr.getPreposition();
+            tbrDTO.noun = tbr.getNoun();
+            expectedDTOs.add(tbrDTO);
+        }
+
+        List<ToBeReviewedDTO> rtnDTOs = reviewSubmittingUserService.getUserPhrasesToBeReviewed(1L);
+        for (ToBeReviewedDTO rtnDTO : rtnDTOs) {
+            for (ToBeReviewedDTO expectedDTO : expectedDTOs) {
+                assertEquals(rtnDTO.hasBeenGroomed, expectedDTO.hasBeenGroomed);
```

Comment:
I believe this only works because both of the test inputs are the same. If they were real-world, two different phrases to be reviewed, at best this would be inefficient, and at worst, it would fail the assertions.

Reply:
Updated with individual, different lists and no looping.